code-collab-app/src/editorpage.js

```diff
@@ -47,36 +30,22 @@ class Editorpage extends React.Component {
     if (!cellContents) {
       cellContents = '';
     }
-
-    this.setState({
-      cellText: this.state.cellText.concat(cellContents),
-      cellLocked: this.state.cellLocked.concat(false),
-    });
-
-    this.setState({
-      cellCount: this.state.cellCount + 1,
-    });
+    this.props.peer_data.add_new_cell(cellContents);
+    // TODO - maybe call peer_data method to update other peers
   }
 
   // Locks a cell so the user cannot edit it
   // key: the index of the cell to lock
   lockEditorCell = (key) => {
     // Lock cell so user may not edit
-    let temp = this.state.cellLocked;
-    temp[key] = true;
-    this.setState({ cellLocked:temp });
+    this.props.peer_data.set_cell_locked(key, true);
   }
 
-  // Update cell text, then Unlock a cell so the user may edit it
+  // Unlock a cell so the user may edit it
   // key: the index of the cell to unlock
-  // updatedCellContents: updated cell text to apply to the cell
-  unlockEditorCell = (key, updatedCellContents) => {
```

Comment:
Here we don't need update cell content because peer_data now have access to everything.